### PR TITLE
Add requests parameters for communication with Girder

### PR DESCRIPTION
**Pull request by pchoisel**
Add a way to specify arguments that will be set to a requests session used for following requests to Girder.

### CODE REVIEW

File: HACKING.md [hunk 1]
```diff
@@ -299,6 +299,22 @@ def some_task(self, file_id):
 
 In this example the producer side REST endpoint creates a token with the scope of the current user, then passes the REST argument `file_id` and the token to the task object's `.delay(...)` function. On the consumer side, the task has access to the GirderClient instance through `self.girder_client`. To get access to `self` the task must be decorated with `@app.task(bind=True)` (See [bound tasks](http://docs.celeryproject.org/en/latest/userguide/tasks.html#bound-tasks) for more information).
 
+##### What is Girder is served over an HTTPS connection ?
```
Comment (zachmullen) [inline HACKING.md:302]:
```suggestion
##### Customizing communication back to Girder server
```

I'm OK with the code changes you've added, but I think we should document it more generally as a feature, i.e. configurability of the request session. `verify: False` could still be used as the example.

Aside from that, it's more accurate to say "What if Girder is served over an _unauthenticated_ TLS connection?" which is kind of a strange use case. What's the reason you need this, out of curiosity?

**Review by zachmullen — CHANGES_REQUESTED** [review]

File: HACKING.md (hunk 1 above)
Comment (pchoisel) [reply HACKING.md:302]: Hi Zach, thanks for the review.

I will document it more tomorrow.
We need this for a project where Girder is not the front-ent webserver (an Apache with load-balancer is set up in front in another VM). However, the security consultants hired by the client require us to cypher the communications between Apache and Girder. To do that, a self-signed certificate is created and we don't want to bother with signing it.

Comment (zachmullen) [reply HACKING.md:302]: That's an... interesting... threat model. Lots of people use tools like [Consul](https://www.consul.io/) for secure inter-service authentication. I'd suggest checking that out if security is something you care about. It manages the hassles of CAs and TLS for you pretty nicely.

Comment (pchoisel) [reply HACKING.md]: Thanks for the suggestion, I'll be sure to suggest it to the client.

I changed my `HACKING.md` section to a feature documentation. Is that what you had in mind ?

File: docs/using-from-girder.rst [hunk 2]
```diff
@@ -64,3 +64,32 @@ be passed into the function in place of the transform object. For example:
 
     def process_file(file):
         return my_task.delay(input_file=GirderFileId(file['_id'])).job
+
+
+Customizing communication back to Girder server
+-----------------------------------------------
+
+Girder Worker is bound to communicate with a Girder instance to update the information about running jobs or download files from the assetstore.  
+In complex platforms, you might want to customize the HTTP communication back to the Girder server.
```
Comment (zachmullen) [inline docs/using-from-girder.rst:73]:
```suggestion
In some scenarios, you might want to customize the HTTP communication back to the Girder server.
```

**Review by zachmullen — CHANGES_REQUESTED** [review]
LGTM after one small tweak

**Comment by pchoisel** [comment]
There are two unit tests failing, but I have no idea why. Would you have a suggestion ?

File: girder_worker/utils.py [hunk 3]
```diff
@@ -207,6 +218,12 @@ def __init__(self, logPrint, url, method=None, headers=None, interval=0.5,
         self._progressCurrent = None
         self._progressMessage = None
 
+        self._session = requests.Session()
```
Comment (zachmullen) [inline girder_worker/utils.py:221]: Can we make sure to close `self._session` in `cleanup`?

**Comment by zachmullen** [comment]
The unit test failures appear to be a result of this branch; I just re-ran them on master and they succeeded.

**Comment by pchoisel** [comment]
I had to modify the tests mocking requests.request
All the tests are passing now and I added the session.close().

**Pull request merged**